api/proto/config/storage.proto

```diff
@@ -242,3 +243,48 @@ message DisksConfig {
   DisksArrayType array_type = 2;
   repeated DisksConfig children = 3;
 }
+
+// SnapshotOpsCmd counter for snapshot cmd
+message SnapshotOpsCmd {
+  uint32 counter = 1;
+  google.protobuf.Timestamp opsTime = 2;
+}
+
+// ZfsSnapshot describes a snapshot for a specific logical
+// volume that must exist on the device. It has a required
+// field volume_uuid for which it was created.
+message ZfsSnapshot {
+  // The real snapshot name in ZFS
+  string id = 1;
+
+  // Display name (User-friendly name). This name does not affect
+  // the snapshot properties in ZFS in any way
+  string display_name = 2;
+
+  // ID of the volume this snapshot belongs to. The field is required
+  // for all commands except for the rename command.
+  string volume_uuid = 3;
+
+  // Time of the last operation on the snapshot
+  uint64 last_cmd_timestamp = 4;
+
+  // Flag indicating that the snapshot has been deleted
+  bool deleted = 5;
+
+  // The command for creating a snapshot, for execution, requires the presence
+  // of a volume_uuid for which the snapshot will be created. By default, populates
+  // display_name with the date the snapshot was taken.
+  SnapshotOpsCmd create = 6;
+
+  // The command to delete a snapshot in ZFS.
+  SnapshotOpsCmd destroy = 7;
```

Comment:
Such flags expected to be inside info messages, not inside configuration. Inside configuration presence of the object with ZfsSnapshot type inside the list indicates that we want to have such object in EVE eventually, absence of the object with ZfsSnapshot type indicates that we want to delete it.